src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/AwsDatasetModification.kt

```diff
@@ -0,0 +1,136 @@
+package com.atlassian.performance.tools.awsinfrastructure.api
+
+import com.atlassian.performance.tools.aws.api.Aws
+import com.atlassian.performance.tools.aws.api.Investment
+import com.atlassian.performance.tools.aws.api.Resource
+import com.atlassian.performance.tools.awsinfrastructure.api.hardware.C5NineExtraLargeEphemeral
+import com.atlassian.performance.tools.awsinfrastructure.api.jira.Jira
+import com.atlassian.performance.tools.awsinfrastructure.api.jira.StandaloneFormula
+import com.atlassian.performance.tools.awsinfrastructure.api.virtualusers.AbsentVirtualUsersFormula
+import com.atlassian.performance.tools.infrastructure.api.dataset.Dataset
+import com.atlassian.performance.tools.infrastructure.api.distribution.PublicJiraSoftwareDistribution
+import com.atlassian.performance.tools.ssh.api.Ssh
+import com.atlassian.performance.tools.workspace.api.RootWorkspace
+import com.atlassian.performance.tools.workspace.api.TestWorkspace
+import org.apache.logging.log4j.LogManager.getLogger
+import java.time.Duration
+import java.util.*
+import java.util.concurrent.TimeUnit
+import java.util.function.Function
+
+class AwsDatasetModification private constructor(
+    private val aws: Aws,
+    private val dataset: Dataset,
+    private val workspace: TestWorkspace,
+    private val newDatasetName: String,
+    private val formula: InfrastructureFormula<*>
+): Function<DatasetTransformation, Dataset> {
```

Comment:
Why did we translate `Consumer<Infrastructure<*>` into `DatasetTransformation`?

Reply:
Because of this discussion:
https://github.com/atlassian/aws-infrastructure/pull/30#discussion_r298488070